analyzers/nodejs/nodejs.go

```diff
@@ -261,6 +263,18 @@ func (a *Analyzer) Analyze() (graph.Deps, error) {
 	return yarn.FromProject(filepath.Join(a.Module.BuildTarget, "package.json"), filepath.Join(a.Module.BuildTarget, "yarn.lock"))
 }
 
+// sanitizeBuildTarget ensures that legacy behavior stays intact but users are warned if it is implemented.
+func (a *Analyzer) sanitizeBuildTarget() {
+	if strings.HasSuffix(a.Module.BuildTarget, "/package.json") {
+		log.Warn("Specifiying the package.json file as a module's target in fossa's config file is no longer supported. Only the directory is neccessary.")
+		a.Module.BuildTarget = strings.TrimSuffix(a.Module.BuildTarget, "/package.json")
```

Comment:
Is it feasible for this to return the build target string instead of modifying in-place? I know it's a lot more plumbing, but avoiding mutation makes code easier to reason about (because it makes data and control flow dependencies explicit).

Reply:
It isn't that much more plumbing at all. This is the way I originally had it structured but decided to get cute. I think that is a better practice and will implement it.